src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -245,6 +245,11 @@ private void AddOriginToResult(string origin, CorsPolicy policy, CorsResult resu
             else if (policy.Origins.Contains(origin))
             {
                 result.AllowedOrigin = origin;
+
+                if(policy.Origins.Count > 1)
```

Comment:
@andrew-laughlin Do you have a repro sample? When making the same request from the second tab, are you using a different origin? 

> serve the cached response up to the second tab, which will fail because it's origin doesn't match the Access-Control-Allow-Origin header.

Do you mean the Response is not empty as it should be for an origin not in Access-Control-Allow-Origin? Failing because the origin is not in Access-Control-Allow-Origin is expected. I am not sure what you mean by "serve the cached response...which will fail".
cc @kichalla